### DISCUSSION TITLE
Webhooks are not working when Attack Challenge Mode is enabled.

### DISCUSSION THREAD

**Discussion by p1ckle-rick** [vercel/community#7221] [Help] [answered]
### Summary

I did some testing and found that when the Attack Challenge Mode is enabled, the webhooks do not work, because the browser verification response is returned instead.

I tried adding a Custom Rule for each webhook and published the changes.

<img width="1233" alt="Screenshot 2024-06-17 at 12 27 16 PM" src="https://github.com/vercel/community/assets/13596285/9073db95-99ef-46c8-9e8f-fb3ce9a469a8">

However, when the Attack Challenge Mode is enabled I still get the browser verification response.

Anything that I am missing?

### Steps to Reproduce

<img width="1233" alt="Screenshot 2024-06-17 at 12 27 16 PM" src="https://github.com/vercel/community/assets/13596285/59d3bb4d-b2ef-4549-afc5-79568bff9ddf">

**Comment by p1ckle-rick** [comment]
@leerob @swarnava @amyegan

Any updates on this?

Is this a bug in Vercel or am I doing some steps incorrectly?

**Comment by leerob** [comment, marked as answer]
This is the expected behavior. Attack Challenge Mode is not compatible with webhooks. If you would like to add challenges more granularly and exclude webhooks, you can create a custom rule for the Firewall to challenge. 

https://vercel.com/docs/security/vercel-waf/custom-rules

**Reply by p1ckle-rick** [reply]
@leerob I did add a custom rule for each webhook path and set the action to `Bypass` but I still get the browser verification response.

**Reply by p1ckle-rick** [reply]
I want the webhooks to go through when the Attack Challenge Mode is on. How do I achieve this?

**Reply by leerob** [reply]
Attack Challenge Mode would take precedence over those rules. You need to _not_ use Attack Challenge Mode, and instead use a custom rule that is a "challenge". That is the same thing as Attack Challenge Mode – a rule with a "challenge" that targets all paths.

[a comment by an automated account is not shown]
